core/src/main/java/com/statful/client/core/sender/BufferedMetricsSender.java

```diff
@@ -316,4 +323,38 @@ private String buildAggregatedUri(final Aggregation aggregation, final Aggregati
 
         return buffersAsList;
     }
+
+    private void scheduleTask(final Runnable runnable) {
+        boolean wasScheduled = false;
+        if (executorService.getQueue().size() < configuration.getMaxWorkerTasksQueueSize()) {
+            Future<?> futureTask = executorService.schedule(runnable, 0, TimeUnit.MILLISECONDS);
+
+            wasScheduled = futuresQueue.offer(futureTask);
+        }
+
+        if (!wasScheduled) {
+            LOGGER.warning("Unable to send metric! The tasks queue is full. Please lower your timeouts or increase the worker thread pool.");
+            cancelAllWaitingTasks();
+        }
+    }
+
+    private void cancelAllWaitingTasks() {
+        helperExecutorService.schedule(tasksKiller(), 0, TimeUnit.MILLISECONDS);
+    }
+
+    private Runnable tasksKiller() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                Collection<Future<?>> tasks = new ArrayList<Future<?>>();
+                futuresQueue.drainTo(tasks);
+
+                for (Future<?> task : tasks) {
+                    task.cancel(false);
```

Comment:
By default, only one worker is running the task and if we cancel all the tasks and leave only one finishing, I think it's ok. I can only see a benefit cancelling already running tasks when there are a lot of workers in the pool.

What do you think?